web-common/src/main/java/edu/unc/lib/boxc/web/common/services/AccessCopiesService.java

```diff
@@ -125,6 +136,41 @@ public String getViewablePdfFilePid(ContentObjectRecord briefObj, AccessGroupSet
         return null;
     }
 
+    /**
+     * Get the first content object that has an original file datastream
+     * and return it if the user has the appropriate permissions
+     * @param briefObj
+     * @param principals
+     * @return
+     */
+    public ContentObjectRecord getContentObject(ContentObjectRecord briefObj, AccessGroupSet principals) {
+        if (permissionsHelper.hasOriginalAccess(principals, briefObj)) {
+            return briefObj;
+        }
+
+        ContentObjectRecord contentObj = getChildFileObject(briefObj, principals);
+        if (contentObj != null && permissionsHelper.hasOriginalAccess(principals, contentObj)) {
+            return contentObj;
+        }
+
+        return null;
+    }
+
+    /**
+     * Get the path of the file that can be downloaded
+     * @param contentObjectRecord
+     * @param principals
+     * @return
+     */
+    public String getDownloadPath(ContentObjectRecord contentObjectRecord, AccessGroupSet principals) {
```

Comment:
`getDownloadUrl` would be more consistent with the naming elsewhere and the content of the returned value.